great_expectations/analytics/anonymizer.py

```diff
@@ -0,0 +1,13 @@
+from __future__ import annotations
+
+import secrets
+from hashlib import md5
+
+
+class Anonymizer:
+    def __init__(self, salt: str | None = None) -> None:
+        self._salt = salt or secrets.token_hex(8)
```

Comment:
I haven't seen where this was used yet, but if we randomize a salt, doesn't our output data become undecryptable/useless?

Reply:
Hmm no that's a fair point - I'm basing this off of the prior implementation but that's something to look into for sure

Reply:
I believe we use the `data_context_id` as salt

Reply:
Then should we make the salt arg non-None?